apis/flowcollector/v1beta1/flowcollector_webhook.go

```diff
@@ -52,6 +52,32 @@ func (r *FlowCollector) ConvertTo(dstRaw conversion.Hub) error {
 	dst.Spec.Loki.Microservices = restored.Spec.Loki.Microservices
 	dst.Spec.Loki.Manual = restored.Spec.Loki.Manual
 
+	if restored.Spec.Agent.EBPF.Advanced != nil {
```

Comment:
Rather than adding more complex conversion logic (sometimes error prone), I find it generally easier to just backport a feature to previous versions. As long as we're not modifying or deleting existing parts of the API, I think it's just fine to backport it to v1beta1?

Reply:
Are you suggesting to add the whole `advanced` sections to every components here ?

Currently in `v1beta1`, only eBPF and flp components have `debug` sections containing a single `Env` map.

Reply:
no that's not what I was suggesting ... iirc `advanced` brought some breaking changes so it had to go via a new API and could not be backported; I was more thinking about adding these fields at the component root like we used to do in beta1, but if you think it won't simplify the things, feel free to keep it like you did